database/postgres/postgres_test.go

```diff
@@ -191,3 +216,53 @@ func TestPostgres_Lock(t *testing.T) {
 			}
 		})
 }
+
+func Test_computeLineFromPos(t *testing.T) {
+	testcases := []struct {
+		pos      uint
+		wantLine uint
+		wantCol  uint
+		input    string
+		wantOk   bool
+	}{
+		{
+			9, 2, 1, "foo bar\nother foo bar", true,
+		},
+		{
+			9, 2, 1, "foo bar\r\nother foo bar", true,
+		},
+		{
+			0, 1, 0, "foo bar\nother foo bar", true,
+		},
+		{
+			6, 1, 6, "foo bar\nother foo bar", true,
+		},
+		{
+			8, 2, 0, "foo bar\nother foo bar", true,
+		},
+		{
+			15, 3, 6, "foo bar\n\nother foo bar", true,
+		},
+		{
+			15, 3, 6, "foo bar\r\n\r\nother foo bar", true,
+		},
+		{
+			999, 0, 0, "foo bar\nother foo bar", false,
+		},
+	}
+	for i, tc := range testcases {
+		t.Run("tc"+strconv.Itoa(i), func(t *testing.T) {
+			gotLine, gotCol, gotOK := computeLineFromPos(tc.input, tc.pos)
+			if gotOK != tc.wantOk {
+				t.Fatalf("expected ok %v but got %v", tc.wantOk, gotOK)
```

Comment:
Use `t.Error()` instead of Fatal. The string formatting isn't necessary, but doesn't hurt either.

Reply:
I use `Fatal` here because I want to stop the test execution to avoid checking things that should not be checked (i.e., if `gotOK` is wrong, then I don't want to check `line` and `col`). The other tests will run because I execute each case in its own test.